components/nexus-repository/src/main/java/org/sonatype/nexus/repository/partial/PartialFetchHandler.java

```diff
@@ -64,8 +64,8 @@ public Response handle(@Nonnull final Context context) throws Exception {
       return response;
     }
 
-    if (!response.getStatus().isSuccessful()) {
-      // We don't interfere with failure responses
+    if (!response.getStatus().isSuccessful() || response.getStatus().getCode() == HttpStatus.PARTIAL_CONTENT) {
```

Comment:
Now that I look them over, I'm not sure it makes sense to partially fetch anything other than 200 - wdyt?